server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -514,7 +548,55 @@ private TsFileProcessor getOrCreateTsFileProcessor(boolean sequence) {
     return tsFileProcessor;
   }
 
-  private TsFileProcessor createTsFileProcessor(boolean sequence)
+  /**
+   * get processor from hashmap, flush oldest processor is necessary
+   *
+   * @param timeRangeId time partition range
+   * @param tsFileProcessorTreeMap tsFileProcessorTreeMap
+   * @param fileList file list to add new processor
+   * @param sequence whether is sequence or not
+   */
+  private TsFileProcessor getOrCreateTsFileProcessorIntern(long timeRangeId,
+      TreeMap<Long, TsFileProcessor> tsFileProcessorTreeMap,
+      Collection<TsFileResource> fileList,
+      boolean sequence)
+      throws IOException, DiskSpaceInsufficientException {
+
+    TsFileProcessor res = null;
+    // we have to ensure only one thread can change workSequenceTsFileProcessors
+    writeLock();
+    try {
+      if (!tsFileProcessorTreeMap.containsKey(timeRangeId)) {
+        // we have to remove oldest processor to control the num of the memtables
+        if (tsFileProcessorTreeMap.size()
+            >= IoTDBConstant.MEMTABLE_NUM_IN_EACH_STORAGE_GROUP / 2) {
+          Map.Entry<Long, TsFileProcessor> processorEntry = tsFileProcessorTreeMap.firstEntry();
+
+          moveOneWorkProcessorToClosingList(sequence, processorEntry.getValue());
```

Comment:
This because the logic of getOrCreateTsFileProcessorIntern is irrelevant with sequence or unsequence. If we don't pass this map as param, we need to use a param named bool sequence, the logic will be much complex. So I pass the map as param.
But for moveOneWorkProcessorToClosingList, the logic of sequence and unsequence is different. It's necessary to use a param bool sequence to control the logic